### DISCUSSION TITLE
Scraping a Website in Supabase and Deno

### DISCUSSION THREAD

**Discussion by dalezak** [supabase/supabase#13117] [Questions]
I'm migrating a Firebase Function to Supabase Edge Functions, however having a difficult time getting `fetch` to work, looking for some advice. 

On Firebase I used `node-fetch` and `cheerio` to scrape a website and everything worked fine, however I've been unable to get it working on Supabase. I tried `fetch` as well as libraries like [axiod](https://deno.land/x/axiod), [dfetch](https://deno.land/x/dfetch), [ky](https://deno.land/x/ky), [summon](https://deno.land/x/summon), [puppeteer](https://deno.land/x/puppeteer), etc but not having much luck. Most are returning `503`, puppeteer is returning [PermissionDenied](https://github.com/lucacasonato/deno-puppeteer/issues/69) error.

What Deno supported library would others recommend to fetch content of a website, so I can pass the HTML into `cheerio`?

**Comment by laktek** [comment]
@dalezak `fetch` should technically work to retrieve content from a web page, unless the site you're trying to access has blocked bot access. `503` response code suggests the server isn't available. Have you verified you can access the site locally using cURL or Postman?

**Comment by dalezak** [comment]
Thanks @laktek, I suspect the request is being blocked as well. However I'm confused why the same code works on Firebase with [node-fetch](https://www.npmjs.com/package/node-fetch) but not on Supabase with `fetch`, [ky](https://deno.land/x/ky), [axiod](https://deno.land/x/axiod), etc. The URL is also accessible because I can load it fine in the browser, and it also works on Firebase.

```javascript
const res = await fetch(url);
const html = await res.text();
const $ = await cheerio.load(html);
```

What's the difference between Firebase Function vs Supabase Edge Functions that would cause the request to be blocked? Perhaps I need to include some request headers like `user-agent`, etc?

I also tried [puppeteer](https://deno.land/x/puppeteer) however can't get past [PermissionDenied: 'Deno.makeTempDir' is not allowed in this context](https://github.com/lucacasonato/deno-puppeteer/issues/69) error. Is it possible to use [puppeteer](https://deno.land/x/puppeteer) with Supabase?

**Comment by dalezak** [comment]
Very strange, yesterday I was getting 503 for every request, however today without any code changes, I am able to fetch html just fine.  Perhaps some changes on Supabase Edge Functions resolved the issue?

**Comment by laktek** [comment]
This is unlikely to be Supabase Edge Functions-related change, but most likely the site you're trying to scrape. They might have had your IP on a banlist (and your IP must have changed), or they could have had your account rate limited before (and rate limit was lifted).
